### DISCUSSION TITLE
Add motion detection toggle control to the camera UI

### DISCUSSION THREAD

**Discussion by sdb9696** [home-assistant/architecture#1139] [Entity Models] [answered]
# Context

The ability to enable or disabled motion detection on `camera` entities has been available as a service for a [long time](https://github.com/home-assistant/core/pull/7961).

Some integrations have created `switch` entities to allow this to be easily changed via the UI. Some of these are:

[Blink](https://github.com/home-assistant/core/blob/dev/homeassistant/components/blink/switch.py)
[Fully kiosk](https://github.com/home-assistant/core/blob/dev/homeassistant/components/fully_kiosk/switch.py)
[Unifi protect](https://github.com/home-assistant/core/blob/dev/homeassistant/components/unifiprotect/switch.py)
[Motion eye](https://github.com/home-assistant/core/blob/dev/homeassistant/components/motioneye/switch.py)
[Hik Vison Cam](https://github.com/home-assistant/core/blob/dev/homeassistant/components/hikvisioncam/switch.py)
[Android IP webcam](https://github.com/home-assistant/core/blob/dev/homeassistant/components/android_ip_webcam/switch.py)
[Tuya](https://github.com/home-assistant/core/blob/dev/homeassistant/components/tuya/switch.py)

However as the `camera` entity already provides the service, it should instead be provided by the front end for the camera entity. This comment from @frenck on a [closed ring PR](https://github.com/home-assistant/core/pull/101250#pullrequestreview-1662676715) make this point.

# Proposal

- Add a toggle control to the camera in the UI to enable/disable motion detection.
- Put the functionality behind a camera feature flag `MOTION_ON_OFF` so that integrations can choose whether to enable it or not.

# UI examples

From this [draft PR](https://github.com/home-assistant/frontend/pull/22010):

## More-info dialogs

(n.b. these are taken from the demo `gallery` pages on the front end so they include the state summary at the top. The actual camera more-info component does not include the state summary)

![image](https://github.com/user-attachments/assets/b2625c02-7e3f-4f6e-9ac5-91f04beebcd4)

![image](https://github.com/user-attachments/assets/772acfe8-f36e-428a-80a0-599bd55ae77f)

## Entity Rows (i.e. device view)

(n.b. should be under Config section)

![image](https://github.com/user-attachments/assets/5cd26a56-c823-4ae9-9f41-1dd2bd1cb47a)

## Motion on/off not supported:

Don't show the toggle if not supported

![image](https://github.com/user-attachments/assets/7ca5c8aa-b4a5-4373-8c60-bb652b31ce20)

# Considerations

- The draft PR does not use the default toggle control in order to distinguish it from a switch that would actually turn the device off/on.
- Currently the UI overrides the the default card on the dashboard to be the `PictureEntity` card. This could be changed to use the entity row including the control and an icon of the picture.

**Comment by MartinHjelmare** [comment, marked as answer]
I'd rather deprecate the motion detection services of the camera entity.

If we disregard the history for a moment, why should this be a feature of the camera entity instead of a separate switch entity?

**Reply by EnochPrime** [reply]
Deprecation makes more sense to me. The motion detection toggle is more of a device configuration than a entity service.

Adding the motion detection in this way could lead into requests for other camera features, such as, AI detections or IR illumination.

**Reply by sdb9696** [reply]
Playing devil's advocate I'd say it's because it's a primary feature of the camera. For motion activated cameras when the motion detection is off the camera is mostly off. I say mostly because other events could still trigger a recording. So it's not really configuration, more like state.

**Reply by EnochPrime** [reply]
So looking through other domains it seems you are asserting that motion detection is similar to climate HVAC mode. Which could be a select domain entity, but is rather built-in to the climate entity model.

**Reply by frenck** [reply]
I agree with Martin on this one; this is not something we should add to the entity component. Instead it can be a regular entity as part of the device.

**Reply by sdb9696** [reply]
@frenck & @edenhaus you have [previously objected](https://github.com/home-assistant/core/pull/101250#pullrequestreview-1662676715) to this on the basis that the camera entity already provides motion detection enable/disable. Do you agree with Martin that we should deprecate the motion detection toggle capability from the camera entity?

N.B. it looks like it's currently implemented in 13 integrations, some of those are in the list above that also implement the switch (tuya, blink, unifiprotect etc)

**Reply by frenck** [reply]
The camera entity provides the state of the motion detection via an attribute. In modern day, we wouldn't add that attribute again.

The camera entity allows for controlling the motion detection (and thus the attribute state of the motion detection) via services.

Your request is to add a switch to the camera platform; which is ambiguous (and probably the confusing part here).

There are no UI controls for cameras a this point; that is not an architectural issue. This can be added using entities, however, unneeded as the current camera entity and available actions support it.

IMHO, there is no reason to add an extra entity nor should be add UI at this point in time; this is mostly because of expected changes to our models in the near future.

The only way around, right now, would be adding a regular switch entity. But.. that is working an UI limitation, of which I'm not a fan.

TL;DR: IMHO, Maybe later but not now.

**Reply by sdb9696** [reply]
With hindsight I probably didn't name this all that well, it was purely a UI proposal to allow toggling the existing motion detection state in the camera platform via the UI. Not adding anything to the platform.

Happy to close this in lieu of the expected model changes. I'll give it a few more days to see if anyone is supportive then close.

**Reply by parautenbach** [reply]
> it was purely a UI proposal to allow toggling the existing motion detection state in the camera platform via the UI. Not adding anything to the platform

If that's the case, my Amcrest doorbell has motion detection, and I notify using events within certain times or under certain circumstances. Having a switch wouldn't help this much, as I may just end up with two automations: One to toggle the switch and the other to notify. 

I think the relevant integrations must just provide the additional entities. For example, my Dahua baby monitor has a switch to enable the IR LED.

**Reply by sdb9696** [reply]
The toggle control wouldn't need to be updated separately. It's not a new switch entity, it's a UI component that is enabling setting the state of the camera's motion detection. Like if you have a service to turn on a light, you will see the light toggle control move to on in the UI. It's the same thing, no new entities.
